front_end_pocket_poll/lib/groups_widgets/group_page.dart

```diff
@@ -461,12 +607,149 @@ class _GroupPageState extends State<GroupPage>
     setState(() {});
   }
 
-  // called when next/back buttons are pressed in event list. Gets the next/previous number of events in group
-  void getNextBatch() {
-    setState(() {
-      this.loading = true;
-    });
-    getGroup();
+  // called when the user scrolls to the bottom of the page
+  Future<void> getNextBatch(
+      final int batchType, final double maxScrollExtent) async {
+    //indicate that we're getting the next batch
+    this.eventTypesToLargestBatchIndexLoaded[batchType]++;
+    final int batchIndex = this.eventTypesToLargestBatchIndexLoaded[batchType];
```

Comment:
This would then just be
final int batchIndex = this.eventTypesToLargestBatchIndexLoaded[batchType+1];

Reply:
yep